Modules/gcmodule.c

```diff
@@ -31,6 +31,63 @@
 #include "pydtrace.h"
 #include "pytime.h"             /* for _PyTime_GetMonotonicClock() */
 
+#define GC_DEBUG (0)  /* More asserts */
+
+// Bit 0 of gc_prev is used for _PyGC_PREV_MASK_FINALIZED in objimpl.h
+#define MASK_COLLECTING              (1 << 1)
+#define MASK_TENTATIVELY_UNREACHABLE (1 << 2)
+
+#define IS_TRACKED(o) (_Py_AS_GC(o)->gc.gc_next != NULL)
+#define IS_TENTATIVELY_UNREACHABLE(o) ( \
+    (_Py_AS_GC(o)->gc.gc_prev & MASK_TENTATIVELY_UNREACHABLE) != 0)
+
+static inline int
+gc_is_collecting(PyGC_Head *g)
+{
+    return (g->gc.gc_prev & MASK_COLLECTING) != 0;
+}
+
+static inline void
+gc_clear_masks(PyGC_Head *g)
+{
+    g->gc.gc_prev &= ~(MASK_COLLECTING | MASK_TENTATIVELY_UNREACHABLE);
+}
+
+static inline Py_ssize_t
+gc_get_refs(PyGC_Head *g)
+{
+    return (Py_ssize_t)(g->gc.gc_prev >> _PyGC_PREV_SHIFT);
+}
+
+static inline void
+gc_set_refs(PyGC_Head *g, Py_ssize_t v)
+{
+    g->gc.gc_prev = (g->gc.gc_prev & ~_PyGC_PREV_MASK)
+        | ((uintptr_t)(v) << _PyGC_PREV_SHIFT);
+}
+
+static inline void
+gc_reset_refs(PyGC_Head *g, Py_ssize_t v)
+{
+    g->gc.gc_prev = (g->gc.gc_prev & _PyGC_PREV_MASK_FINALIZED)
+        | MASK_COLLECTING
+        | ((uintptr_t)(v) << _PyGC_PREV_SHIFT);
+}
+
+static inline void
+gc_set_prev(PyGC_Head *g, PyGC_Head *v)
+{
+    g->gc.gc_prev = (g->gc.gc_prev & ~_PyGC_PREV_MASK)
```

Comment:
This is the same as `_PyGCHead_SET_PREV()`, right?

Reply:
There is small difference; `_PyGCHead_SET_PREV(g, v)` assumes `v` is pure pointer (no masks).
But only caller of this function passes pure pointer.  So `_PyGCHead_SET_PREV()` can be used instead.
I'll remove this function.